packages/react-test-renderer/src/ReactTestRendererAct.js

```diff
@@ -42,18 +42,20 @@ function flushEffectsAndMicroTasks(onDone: (err: ?Error) => void) {
 
 function act(callback: () => Thenable) {
   let previousActingUpdatesScopeDepth;
+  let previousActingUpdatesSigil;
   if (__DEV__) {
     previousActingUpdatesScopeDepth = actingUpdatesScopeDepth;
+    previousActingUpdatesSigil = ReactShouldWarnActingUpdates.current;
     actingUpdatesScopeDepth++;
-    ReactShouldWarnActingUpdates.current = true;
+    // we use the function flushPassiveEffects directly as the sigil,
+    // since it's unique to a renderer
+    ReactShouldWarnActingUpdates.current = flushPassiveEffects;
```

Comment:
Let's use an empty object that's local to this module instead.

Reply:
ReactFiberScheduler wouldn't be able to read it when it needs to compare though? in `warnIfNotCurrentlyActingUpdatesInDev` 
https://github.com/facebook/react/blob/7f3fd8f8667f75821e02037aa3c229fd942baf39/packages/react-reconciler/src/ReactFiberScheduler.js#L2086